libraries/WiFiClientSecure/src/WiFiClientSecure.h

```diff
@@ -0,0 +1,63 @@
+/* Provide SSL/TLS functions to ESP32 with Arduino IDE
+ * by Evandro Copercini - 2017 - Apache 2.0 License
+ */
+
+#ifndef WiFiClientSecure_h
+#define WiFiClientSecure_h
+#include "Arduino.h"
+#include "IPAddress.h"
+#include <WiFi.h>
+
+#include "ssl_client.h"
+
+struct  ssl_context;
+
+class WiFiClientSecure : public Client {
+
+public:
+
+	WiFiClientSecure();
+	WiFiClientSecure(uint8_t sock);
+
+	virtual int connect(IPAddress ip, uint16_t port);
+	virtual int connect(const char *host, uint16_t port);
+	virtual size_t write(uint8_t data );
+	virtual size_t write(const uint8_t *buf, size_t size);
+	virtual int available();
+	virtual int read();
+	virtual int read(uint8_t *buf, size_t size);
+	virtual int peek();
+	virtual void flush();
+	virtual void stop();
+	virtual uint8_t connected();
+	virtual operator bool();
+
+	void setCACert(unsigned char *rootCA);
+	void setCertificate(unsigned char *client_ca);
+	void setPrivateKey (unsigned char *private_key);
+
+	int connect(IPAddress ip, uint16_t port, unsigned char* rootCABuff, unsigned char* cli_cert, unsigned char* cli_key);
+	int connect(const char *host, uint16_t port, unsigned char* rootCABuff, unsigned char* cli_cert, unsigned char* cli_key);
+
+	int startClient(sslclient_context *ssl_client, uint32_t ipAddress, uint32_t port, unsigned char* rootCABuff, unsigned char* cli_cert, unsigned char* cli_key);

```

Comment:
all those methods should be private/protected right? The user does not have access to the sslclient_context anyway.